portfolio/src/main/webapp/assets/js/testimonials.js

```diff
@@ -0,0 +1,111 @@
+  }
+}
+
+function createTestimonialElement(text,author,title){
+    // // create elements within block quote	
+    const blockQuoteParagraphElement = document.createElement('p');	
+    const iTagElement = document.createElement('i');
+    iTagElement.setAttribute("class","fas fa-quote-left");
+    const blockQuoteText = document.createTextNode(text);
+
+    // create block quote wrapper and add all elements to it
+    const blockQuoteElement = document.createElement('blockquote');
+    blockQuoteElement.setAttribute("class", "quote");
+    blockQuoteParagraphElement.appendChild(iTagElement);
+    blockQuoteParagraphElement.appendChild(blockQuoteText);
+    blockQuoteElement.appendChild(blockQuoteParagraphElement);
+
+    // create span elements within author attribution
+    const authorSpanElement = document.createElement("span");
+    authorSpanElement.setAttribute("class","name");
+    authorSpanElement.appendChild(document.createTextNode(author));
+    const titleSpanElement = document.createElement("span");
+    titleSpanElement.setAttribute("class","title");
+    titleSpanElement.appendChild(document.createTextNode(title));
+    
+    //add span and other elements to paragraph element in author attribution
+    const paragraphElement = document.createElement('p');
+    paragraphElement.setAttribute("class","source");	
+    paragraphElement.appendChild(authorSpanElement);
+    paragraphElement.appendChild(document.createElement('br'));
+    paragraphElement.appendChild(titleSpanElement);
+
+    // wrap elements into a div and return
+    testimonialElementLi = document.createElement('li');
+    testimonialElementLi.appendChild(blockQuoteElement);
+    testimonialElementLi.appendChild(paragraphElement);
+
+    return testimonialElementLi;
+}
+
+function filterFunction() {
+    var input, filter, ul, li, a, i, txtValue;
+    input = document.getElementById("filterInput");
+    filter = input.value.toUpperCase();
+    ul = document.getElementById("testimonialUl");
+    li = ul.getElementsByTagName("li");
+    for (i = 0; i < li.length; i++) {
+        blockQuote = li[i].getElementsByTagName("blockquote")[0];
+        p = blockQuote.getElementsByTagName("p")[0];
+        txtValue = p.textContent || p.innerText;
+        if (txtValue.toUpperCase().indexOf(filter) > -1) {
+            li[i].style.display = "";
+        } else {
+            li[i].style.display = "none";
+        }
+    }
+}
+
+function startListening(){
+    annyang.resume();
+    console.log("mouse down");
+}
+
+function stopListening(){
+    annyang.pause();
+    console.log("mouse up");
+}
+
+function initializeVoiceControl(){
+    if (annyang) {
+        var typeThis = function(spoken){
+            // OPTION A Works when going back and forth between typing and voice
+            var textElement = document.getElementById('text');
+            textElement.textContent += spoken;
+            textElement.value += spoken;
```

Comment:
IMO this line is the correct way of doing things. HTML inputs all have a readable/writable `value` for their contents. I also think you should remove the line above, it looks to me like it would work without it.

Reply:
okay that is very helpful, thank you! Which line above are you referring to?

Reply:
The one setting textContent.

Reply:
got it- thanks